File: src/python/keyboard_request.py
import request_generator


class KeyboardRequest(request_generator.RequestGenerator):
    def __init__(self, client):
        super(KeyboardRequest, self).__init__(client)
        self.type = 'action'
        self.device = 'keyboard'

    def type(self, keys):
        self.action = 'type'
        self.params = keys
        return self.send_request()

    def type_string(self, strings):
        self.action = 'type_string'
        self.params = strings
        return self.send_request()

    def combination(self, keys):
        self.action = 'combination'
        self.params = keys
        return self.send_request()

File: src/natives/csharp/IPC/KeyboardRequest.cs
using Repeat.ipc;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Repeat.IPC {
    class KeyboardRequest : RequestGenerator {

        public KeyboardRequest(RepeatClient client) : base(client) {
            this.Type = "action";
            this.Device = "keyboard";
        }

        public bool DoType(params int[] keyCodes) {
            Action = "type";
            ClearParams();
            ParamInt.AddRange(keyCodes);
            return SendRequest();
        }

        public bool DoType(params string[] strings) {
            Action = "type_string";
            ClearParams();
            ParamStrings.AddRange(strings);
            return SendRequest();
        }

        public bool Combination(params int[] keyCodes) {
            Action = "combination";
            ClearParams();
            ParamInt.AddRange(keyCodes);
            return SendRequest();
        }
    }
}
